### ISSUE TITLE
Building: Stairs

### ISSUE THREAD

**Issue by garrynewman** [Facepunch/rust-issues#100]
Look into an alternative stair block. Possibly an L shape.

**Comment by OverlordJ19** [comment]
I think that would be cool, but how many blocks would it take up? 1 is prefered, but 3 is probably the only feasible way. However we could do that ourselves with 2 stairs and a 1/2 block.

Also, have you guys considered adding ladders? They would be extremely helpful as well as super cool.
Maybe even implementing Siege weapons, ways to get into bases from on top without having to build a super tower and jump in.

**Comment by paroxum** [comment]
L shape sounds good. I'm guessing Nobody wants a U shaped one? Helk, what's your preference compared to legacy?

**Comment by garrynewman** [comment]
Try it out I guess, if they both work we can add both. I was worried that the U shape would be too tight.

**Comment by paroxum** [comment]
First one is one meter wide, benefits of 1m wide flight of stairs, the gradient is lower and looks more realistic. Chain staircases will look more spacious and less cramped.
Cons are you cannot have two players cross each other on this stair.
![image](https://cloud.githubusercontent.com/assets/11387355/7199039/525c296e-e4e9-11e4-9443-837addd245b0.png)

Second one is 1.5m wide flight of stairs, cons: the gradient is unrealistic at 45degree but the player can still climb it perfectly fine. Advantages: two players can be on it fine at the same time, if the player bounding box is not wider than 0.75m. 
![image](https://cloud.githubusercontent.com/assets/11387355/7199047/62efc31c-e4e9-11e4-9b31-b75b6cf33f4c.png)

Tbh, I didn't think the player would be able to climb the 45 slope but since it can, it might be the best solution for gameplay. Nobody likes getting stuck in stairs. Esp when we add railings later.

**Comment by garrynewman** [comment]
Good research. I think ultimately the bottom will be better. The gradient will probably feel less steep once they're stairs and not a slope.

**Comment by paroxum** [comment]
Indeed, I did start a mockup now using the stair meshes in max, and it looks good. I'll move forward on this.

**Comment by paroxum** [comment]
![image](https://cloud.githubusercontent.com/assets/11387355/7199949/74936af4-e4f0-11e4-81ef-8967ae5e4971.png)

**Comment by paroxum** [comment]
![stairs_railings](https://cloud.githubusercontent.com/assets/11387355/7202329/42d10300-e509-11e4-8069-8d1706f89306.jpg)

**Comment by AlexanderLaheij** [comment]
Those stairs , :D

**Comment by paroxum** [comment]
I did commit on main with models and collisions for this stair. I also created a prefab variant for the lshape stair block based off the block.halfheight.slanted and retargetted to the new models/collisions. 

I don't really know how to make the sockets work for the railings at the moment so garry have fun with that when you jump on to this. I'll move onto the other skin types for the other models to make.

**Comment by Helkus** [comment]
Not feeling the banisters
from an aesthetic standpoint they're way too perfect and from a gameplay perspective they would lock players in and limit movement and I would image be annoying for players to deal with from a movement perspective, other than that I love it

**Comment by paroxum** [comment]
That's why we will be making them optional as the players will chose whether they place them or not.

**Comment by paroxum** [comment]
I can do a pass later to add chaos in the way the banisters are made with offset planks, or slashed/missing bars etc.

**Comment by paroxum** [comment]
I'd rather do that than having something blocky personally.

**Comment by Tuntenfisch** [comment]
![segmented stair](https://cloud.githubusercontent.com/assets/11965324/7206134/bde3baf4-e52e-11e4-8cfb-69eb8b1d2b72.png)

you might want to take into consideration dividing the whole stair component (as highlighted above)  into 2 separate parts. The platform (highlighted red) is 1.5 * 1.5 (which is a quarter foundation). The stair (highlighted green) is 1.5 * 1.5 as well. 


Making them separate would allow for some neat things, like getting up a half block with a single stair component (as seen below) 

![example 2](https://cloud.githubusercontent.com/assets/11965324/7206170/145496b0-e52f-11e4-853a-6be74986db73.png)

Alternatively building this stair (as seen below) would still leave half the foundation open and be one foundations in length.

![example 1](https://cloud.githubusercontent.com/assets/11965324/7206164/04360b60-e52f-11e4-896e-01447b48f097.png)

The platform (highlighted red) would also be beneficial for ladders (if they are planned) which could only occupy quarter of a foundation, leaving 3 quarters of free room to put those platforms.

**Comment by garrynewman** [comment]
Because the building system isn't complicated enough yet

**Comment by y0himba** [comment]
How will these fit into a building? How hard are they going to be to place and how much space will they occupy?

**Comment by Tuntenfisch** [comment]
@garrynewman I don't want to spam this issue with more lengthy posts so I'll rather link to my response on reddit: https://www.reddit.com/r/playrust/comments/32xbew/new_stairs/cqfpkhj

Cheers!

**Comment by arehberg** [comment]
Saw this mentioned on reddit and thought it was worth bringing up here: Will these line up properly with doors?

**Comment by paroxum** [comment]
No, they wont, and they cant. Players that absolutely want their door at the end of their staircases will have to plays the acrobats just like it was before.

**Comment by AlexanderLaheij** [comment]
![rust123123](https://cloud.githubusercontent.com/assets/11994549/7210524/ba01826c-e552-11e4-8c25-45e990566c6a.jpg)
@garrynewman 's team . So, those red-highlighted staircase elements, would basically be quarter sized ceilings, which would allow from some intresting stair building. Those 1/4 sized ceilings would also be great for a neat balcony inside or outside a player's base (build creation). Picture explains it all.

**Comment by garrynewman** [comment]
@AlexanderLaheij I feel it adds a lot of complexity that doesn't add a lot of benefit.

**Comment by T-im** [comment]
The quarter floor/ceiling parts could bring a lot of variation. At the moment, houses are pretty bulky. These smaller floors could function as a number of things; 
- A more subtle balcony, like mentioned above. But they could also go on the inside of an outer wall, as a walkway to scout the perimeter, that doesn't block off all the light if the space between the wall and the base is only 1 foundation.
- Shelves. currently a floor at half-height takes up a lot of your base, but it's useful for putting multiple large chests in a 1x1 space. Imagine a 1 foundation wide hallway from one room to another, with one wall lined with chests (2 per foundation).
- More customizable stairs like suggested earlier.
- Make-shift tables, although those will become obsolete once the tables from the concept-art are implemented.

The raiders of Rust won't care much. The builders in Rust will rejoice at every additional customization they can add to their bases.

Of course, the benefit of full width foundations/triangular foundations, is that you only need 1 wall/doorframe/window/railing length. Implementing smaller (or larger, e.g. barn doors) width building pieces would require an amount of coding.

**Comment by paroxum** [comment]
I did an update to our railings after a discussion I had with Helk about them. These new ones should look more makeshift/chaotic.

![stairs_railings2](https://cloud.githubusercontent.com/assets/11387355/7226779/8e9f0b88-e743-11e4-8928-ee03d05c8e24.jpg)
